module/VuFind/src/VuFind/Config/PathResolverFactory.php

```diff
@@ -82,14 +85,37 @@ public function __invoke(
         if (!empty($options)) {
             throw new \Exception('Unexpected options sent to factory.');
         }
-        $localDirs = defined('LOCAL_OVERRIDE_DIR')
+        $localDirs = [];
+        $currentDir = defined('LOCAL_OVERRIDE_DIR')
             && strlen(trim(LOCAL_OVERRIDE_DIR)) > 0
-                ? [
-                    [
-                        'directory' => LOCAL_OVERRIDE_DIR,
-                        'defaultConfigSubdir' => $this->defaultLocalConfigSubdir,
-                    ],
-                ] : [];
+            ? LOCAL_OVERRIDE_DIR : '';
+        while (!empty($currentDir)) {
+            $systemConfigFile = $currentDir . '/DirLocations.ini';
+            $systemConfig = new Config(
+                file_exists($systemConfigFile)
+                    ? $this->getIniReader()->fromFile($systemConfigFile)
+                    : []
+            );
+            array_unshift(
+                $localDirs,
+                [
+                    'directory' => $currentDir,
+                    'defaultConfigSubdir' =>
+                        $systemConfig['Local_Dir']['config_subdir']
+                        ?? $this->defaultLocalConfigSubdir,
+                ]
+            );
+
+            $parentDir = '';
+            if (!empty($systemConfig['Parent_Dir']['path'])) {
+                $parentDir = $systemConfig['Parent_Dir']['path'];
+                if ($systemConfig['Parent_Dir']['is_relative_path']) {
```

Comment:
This needs a default value to avoid an unset configuration triggering a notice. I assume `?? false` is what you intend. Maybe this whole block could be simplified to:

```
            // If there's a parent, set it as the current directory for the next loop iteration:
            if (!empty($systemConfig['Parent_Dir']['path'])) {
                $isRelative = $systemConfig['Parent_Dir']['is_relative_path'] ?? false;
                $parentDir = $systemConfig['Parent_Dir']['path'];
                $currentDir = $isRelative ? $currentDir . '/' . $parentDir : $parentDir;
            } else {
                $currentDir = '';
            }
```

Reply:
Yes, i think your suggestion look much cleaner. I have taken this.